### ISSUE TITLE
is it possible to have a global error handler for actions?

### ISSUE THREAD

**Issue by jayarjo** [mobxjs/mobx-state-tree#829]
While it is possible to write an error handler for each and every action, it could be useful to be able to declare one global error handler (catcher).

**Comment by mweststrate** [comment]
This can easily achieved by adding middleware to the root of your tree,
which run's `next()` inside a try-catch. See the middleware docs for more
details

Op do 24 mei 2018 om 06:30 schreef Davit Barbakadze <
notifications@github.com>:

**Comment by jayarjo** [comment]
I tried that, but every now and then, global try/catch hits: `Cannot read property 'done' of undefined` exception. From what I see it happens if the action that is invoked with `next()` internally throws an exception itself.

**Comment by adnan-i** [comment]
There is another issue discussing the same problem (#933), but even though both these issues are closed, none provided much help in dealing with this problem. I feel it's redundant to open a third ticket talking about the same exact issue.
I've also searched the spectrum chat, but without success.
I'm thankful for this great library, and I hate bothering open-source developers with usage questions. I've tried all I could (including suggestions from @mweststrate in this issue and issue #933) but global error handler solution still evades me.
Is there any chance we could be provided with a working example of a global handler middleware that can catch any error thrown in either sync or flow actions?
I believe many users would be delighted for not having to wrap each and every action in try/catch blocks.
